### PR TITLE
Invalidate dependency cache of all whelk instances on write

### PR DESCRIPTION
**Pull request by olovy**
Send a notification to all other whelk instances to invalidate affected parts of DependencyCache when a document is saved. PostgreSQLs LISTEN/NOTIFY is used as a simple IPC mechanism.

This uses one more physical database connection per whelk instance.

Here is a previous iteration that instead sends the ids of the previous and current document versions and lets each listener calculate which links have been added and removed https://github.com/libris/librisxl/compare/feature/clear-deps-cache

### CODE REVIEW

**Comment by olovy** [comment]
Ping @lrosenstrom @niklasl @andersju

File: whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy [hunk 1]
```diff
@@ -2554,6 +2568,98 @@ class PostgreSQLComponent {
             }
         }
     }
+    
+    enum NotificationType {
+        dependency_cache_invalidate // LISTEN silently fails to register if upper case(???)
```
Comment (andersju) [inline whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy:2573]: Given https://stackoverflow.com/a/5173993 I'm guessing the channel name here is _not_ forcibly lowercased
```
try (PreparedStatement statement = getMyConnection().prepareStatement('SELECT pg_notify(?, ?)')) {
    statement.setString(1, type.toString())
```
while here it _is_
```
statement.execute("LISTEN ${t.toString()}")
```
?

**Review by andersju — APPROVED** [review]
Nice! LGTM. Let's try it on dev!

File: whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy (hunk 1 above)
Comment (olovy) [reply whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy:2573]: Ah!

I'll make that clear.

Comment (olovy) [reply whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy:2573]: Forgot to push it here....
On `develop` now https://github.com/libris/librisxl/commit/8a047738c4ae3ac9cbfd75d209009a6cd697fef8

**Pull request merged**